src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -174,12 +212,14 @@ QgsMeshDatasetValue QgsMeshLayer::datasetValue( const QgsMeshDatasetIndex &index
 {
   QgsMeshDatasetValue value;
 
-  if ( mTriangularMesh && dataProvider() && dataProvider()->isValid() && index.isValid() )
+  const QgsTriangularMesh *mesh = triangularMesh();
```

Comment:
better to rename local/member variables && fuctions to to baseTriangularMesh, simplifiedTriangularMesh etc, use it consistenty in the code to improve readability

Reply:
`triangularMesh()` without parameters return the base triangular mesh as specified if the doc related to this method. Maybe you want a method that return the baseTriangularMesh. This method exists but its aim is mainly to construct a base triangular mesh if needed... It will be confusing to provide a new function...

Reply:
ok